src/uct/ib/mlx5/dc/dc_mlx5_ep.h

```diff
@@ -383,10 +383,10 @@ uct_dc_mlx5_iface_dci_can_alloc(uct_dc_mlx5_iface_t *iface, uint8_t pool_index)
 }
 
 static UCS_F_ALWAYS_INLINE void
-uct_dc_mlx5_iface_dcis_array_copy(void *dst, const void *src, size_t length)
+uct_dc_mlx5_iface_dcis_array_copy(void *dst, void *src, size_t length)
 {
-    const uct_dc_dci_t *src_dcis = (const uct_dc_dci_t*)src;
-    uct_dc_dci_t *dst_dcis       = (uct_dc_dci_t*)dst;
+    uct_dc_dci_t *src_dcis = (uct_dc_dci_t*)src;
```

Comment:
1. keep it const*. we ca do
```
ucs_queue_head_t tmp_q;
...
tmp_q = src_dcis[i].txqp.outstanding;
ucs_queue_splice(&dst_dcis[i].txqp.outstanding, &tmp_q);
```

Reply:
That will cause a segfault as `outstanding` will point to `tmp_q`, but `tmp_q` will be dead at the end of this scope

Reply:
outstanding should not point to tmp_q after splice